Spanner/src/Numeric.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner;
+
+/**
+ * Represents a value with a data type of
+ * [Numeric](https://cloud.google.com/spanner/docs/data-types#numeric_type).
+ *
+ * It supports a fixed 38 decimal digits of precision and 9 decimal digits of scale, and values
+ * are in the range of -99999999999999999999999999999.999999999 to
+ * 99999999999999999999999999999.999999999.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Spanner\SpannerClient;
+ *
+ * $spanner = new SpannerClient();
+ *
+ * $numeric = $spanner->numeric('99999999999999999999999999999999999999.999999999');
+ * ```
+ */
+class Numeric implements ValueInterface
+{
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * @param string|int|float $value The NUMERIC value.
+     * @throws \InvalidArgumentException
+     */
+    public function __construct($value)
+    {
+        $value = (string) $value;
+        // allow minus sign at the beginning
```

Comment:
Curious about this not supporting '+' at the beginning which I think is explicitly allowed by the backend. The backend will strip the '+' when it stores it. So since you have a choice here, should we support it?

In Node.js for example, I can't support it because the underlying library we're using doesn't support it.

Reply:
`bcmath` appears to support '+' prefixed string so I'm willing to add support for it.

@jdpedrie Any concerns with this?

Reply:
No issues, but let's make sure there's system test coverage for it!